### PR TITLE
tegra_pcie: use switch instead of if

### PR DESCRIPTION
**Pull request by ferhatgec**
looks much better.

### CODE REVIEW

**Comment by bsdimp** [comment]
Why not the simpler table lookup:
```
static bus_size_t
tegra_pcib_pex_ctrl(struct tegra_pcib_softc *sc, int port)
{
        bus_size_t rv[TEGRA_PCIB_MAX_PORTS] = {
                AFI_PEX0_CTRL, AFI_PEX0_CTRL, AFI_PEX0_CTRL };

        if (port >= TEGRA_PCIB_MAX_PORTS || port < 0)
                panic("invalid port number: %d\n", port);

        return rv[port];
}
```

File: sys/arm/nvidia/tegra_pcie.c [hunk 1]
```diff
@@ -837,15 +837,17 @@ tegra_pcib_pex_ctrl(struct tegra_pcib_softc *sc, int port)
...
 		return (AFI_PEX1_CTRL);
-	else if (port == 2)
-		return (AFI_PEX2_CTRL);
-	else
-		panic("invalid port number: %d\n", port);
+	case 2:
+		return (AFI_PEX_2_CTRL);
+	}
+	
+	panic("invalid port number: %d\n", port);
```
Comment (semihalf-wojtas-marcin) [inline sys/arm/nvidia/tegra_pcie.c:850]: Can you please put it explicitly under `defailt: ` ?

**Comment by ferhatgec** [comment]
It's looks good right now.

File: sys/arm/nvidia/tegra_pcie.c [hunk 2]
```diff
@@ -835,17 +835,16 @@ tegra_pcib_msi_map_msi(device_t dev, device_t child, struct intr_irqsrc *isrc,
...
-		return (AFI_PEX2_CTRL);
-	else
-		panic("invalid port number: %d\n", port);
+	switch (port) {
+	case 0:
+                return (AFI_PEX0_CTRL);
+	case 1:
+                return (AFI_PEX1_CTRL);
+	case 2:
+                return (AFI_PEX_2_CTRL);
```
Comment (semihalf-wojtas-marcin) [inline sys/arm/nvidia/tegra_pcie.c:844]: Some undesired whitespace change slipped through - tabs are replaced by spaces in all "return" + "panic" lines.

File: sys/arm/nvidia/tegra_pcie.c [hunk 3]
```diff
@@ -835,17 +835,16 @@ tegra_pcib_msi_map_msi(device_t dev, device_t child, struct intr_irqsrc *isrc,
...
+	case 0:
 		return (AFI_PEX0_CTRL);
-	else if (port == 1)
+	case 1:
 		return (AFI_PEX1_CTRL);
-	else if (port == 2)
-		return (AFI_PEX2_CTRL);
-	else
+	case 2:
+		return (AFI_PEX_2_CTRL);
```
Comment (semihalf-wojtas-marcin) [inline sys/arm/nvidia/tegra_pcie.c:844]: Does it compile? AFI_PEX2_CTRL changed to AFI_PEX_2_CTRL and PEX0/PEX1 equivalents remained intact.

Comment (ferhatgec) [reply sys/arm/nvidia/tegra_pcie.c]: This is gonna to be a journey. :')

**Review by semihalf-wojtas-marcin — APPROVED** [review]
LGTM, thanks.

**Comment by bsdimp** [comment]
> LGTM, thanks.

Would you like me to merge it in? Or would you rather do that?

**Comment by semihalf-wojtas-marcin** [comment]
Patch merged: https://cgit.freebsd.org/src/commit/?id=8df71ea1aa3b39f84e7ee3122a333caab57c3e1a

Thanks @ferhatgec !

**Pull request closed without merging**